cli/version_check.go

```diff
@@ -13,7 +13,7 @@ import (
 
 // The terraform --version output is of the format: Terraform v0.9.5-dev (cad024a5fe131a546936674ef85445215bbc4226+CHANGES)
 // where -dev and (commitid+CHANGES) is for custom builds or if TF_LOG is set for debug purposes
-var TERRAFORM_VERSION_REGEX = regexp.MustCompile("Terraform (v?[\\d\\.]+)(?:-dev)?(?: .+)?")
+var TERRAFORM_VERSION_REGEX = regexp.MustCompile("Terraform (v?[\\d\\.]+)(?:.*)")
```

Comment:
Looking at this again, I don't think this regex is quite correct. The `[\\d\\.]+` portion accepts any combination of digits or dots. E.g., `v33333`, `v.....`, and `v3....3......3` would all be valid. Also, two NITs: the final `(?:.*)` can be simplified to just `.*`. and if we use backticks to wrap the string instead of double quotes, we won't need nearly as much escaping. 

Putting that all together, we probably want a regex that looks something more like this:

```go
var TERRAFORM_VERSION_REGEX = regexp.MustCompile(`Terraform (v?(?:\d\.)*\d).*`)
```

Reply:
I think since terraform [adheres](https://www.terraform.io/docs/extend/best-practices/versioning.html) to semver it may be cleaner to check for `MAJOR.MINOR.PATCH` numbers specifically rather than rely on a quantifier.